### PR TITLE
refactor: remove trailing usage of @SubstrateTest

### PR DESCRIPTION
**Pull request by machi1990**
follows up https://github.com/quarkusio/quarkus/pull/5007

/cc @gwenneg

### CODE REVIEW

File: independent-projects/tools/common/src/test/resources/templates/basic-rest/kotlin/native-test-resource-template.ftl
Comment (gsmet) [inline independent-projects/tools/common/src/test/resources/templates/basic-rest/kotlin/native-test-resource-template.ftl]: This and a few others below looks wrong. `Test` is repeated.

**Review by gsmet — CHANGES_REQUESTED** [review]

File: independent-projects/tools/common/src/test/resources/templates/basic-rest/kotlin/native-test-resource-template.ftl
Comment (machi1990) [reply independent-projects/tools/common/src/test/resources/templates/basic-rest/kotlin/native-test-resource-template.ftl]: Good catch.

Comment (machi1990) [reply independent-projects/tools/common/src/test/resources/templates/basic-rest/kotlin/native-test-resource-template.ftl]: done

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Looks better, I'll let @gwenneg review.

File: integration-tests/test-extension/src/test/java/io/quarkus/it/extension/FinalFieldReflectionInGraalITCase.java
Comment (gwenneg) [inline integration-tests/test-extension/src/test/java/io/quarkus/it/extension/FinalFieldReflectionInGraalITCase.java]: This test is still annotated with `@SubstrateTest` to make sure we don't break the deprecated classes until their removal. We should probably keep it that way for now and add a comment to explain why it is not annotated with `@NativeImageTest`.

File: integration-tests/main/src/test/java/io/quarkus/it/main/HealthCheckTestCase.java
Comment (gwenneg) [inline integration-tests/main/src/test/java/io/quarkus/it/main/HealthCheckTestCase.java]: I don't know the reason, but this test was designed to prevent it from being launch in native mode. That's why the presence of both `@SubstrateTest` and `@NativeImageTest` is checked here. It should probably stay that way until `@SubstrateTest` is removed.

**Comment by gwenneg** [comment]
The build failure seems unrelated.

File: integration-tests/main/src/test/java/io/quarkus/it/main/HealthCheckTestCase.java
Comment (machi1990) [reply integration-tests/main/src/test/java/io/quarkus/it/main/HealthCheckTestCase.java]: The reason is because injection is not supported in native mode. I think we can safely remove Substrate here, if it is launched as native mode, the test will fail.

File: integration-tests/test-extension/src/test/java/io/quarkus/it/extension/FinalFieldReflectionInGraalITCase.java
Comment (machi1990) [reply integration-tests/test-extension/src/test/java/io/quarkus/it/extension/FinalFieldReflectionInGraalITCase.java]: Good to know this. 
```
// Use @SubstrateTest to ensure backward compatibility. 
// TODO replace with @NativeImageTest once SubstrateTest is removed. 
```

WDYT?

Comment (gwenneg) [reply integration-tests/test-extension/src/test/java/io/quarkus/it/extension/FinalFieldReflectionInGraalITCase.java]: That sounds good!

File: integration-tests/main/src/test/java/io/quarkus/it/main/HealthCheckTestCase.java
Comment (gwenneg) [reply integration-tests/main/src/test/java/io/quarkus/it/main/HealthCheckTestCase.java]: That's good to know, but this reason might not be obvious to everyone. I'd rather keep these two checks and add an explicit message in case of failure. And also a comment explaining why we're testing `@SubstrateTest` :) That way, we'd be sure nobody will ever try to run this test in native mode even if the injection is failing.

Comment (gwenneg) [reply integration-tests/main/src/test/java/io/quarkus/it/main/HealthCheckTestCase.java]: Actually, we should probably use `@DisabledOnNativeImage` instead.

Comment (machi1990) [reply integration-tests/main/src/test/java/io/quarkus/it/main/HealthCheckTestCase.java]: It is minor but I still think we can remove the `@SubstrateTest` check here: I do not think we'll try to run this test in native mode with `@SubstrateTest` - since it is deprecated. 

- I can however add a comment - 
```
The Injection test is not meant to be ran in native mode as it is not supported - https://quarkus.io/guides/getting-started-testing#native-executable-testing
```

**Review by gwenneg — APPROVED** [review]
LGTM, thanks!

**Comment by gwenneg** [comment]
The build failure was caused by a `FinalFieldReflectionInGraalITCase` imports sort issue. Could you fix that please @machi1990?

File: independent-projects/tools/common/src/test/resources/templates/basic-rest/kotlin/native-test-resource-template.ftl
Comment (Sanne) [reply independent-projects/tools/common/src/test/resources/templates/basic-rest/kotlin/native-test-resource-template.ftl]: wasn't fixed yet? :)

**Review by Sanne — CHANGES_REQUESTED** [review]
Wrong name `io.quarkus.test.junit.NativeImageTestTest` in the templates

**Comment by machi1990** [comment]
> Wrong name `io.quarkus.test.junit.NativeImageTestTest` in the templates

Good catch. These ones should have been fixed earlier though ;-)

Thanks @Sanne

File: independent-projects/tools/common/src/test/resources/templates/basic-rest/kotlin/native-test-resource-template.ftl
Comment (machi1990) [reply independent-projects/tools/common/src/test/resources/templates/basic-rest/kotlin/native-test-resource-template.ftl]: I might have walked on my feet myself after changing machines :-(

Thanks for catching this.

**Review by Sanne — APPROVED** [review]

**Comment by Sanne** [comment]
NP, thanks! Let's wait for CI one more time.

**Comment by machi1990** [comment]
CI failure tracked by #5536 and is unrelated to the change.

Merging. Thanks @gwenneg @gsmet @Sanne

**Pull request merged**
